### PR TITLE
json-extension: use an unique token for progress example

### PR DESCRIPTION
**Pull request by perrinjerome**
The spec [1] and pygls implementation [2] need the token to be unique.

By using an hardcoded token like this, the progress works only one time,
the second time there's an error that token is already registered.

[1]: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#serverInitiatedProgress

[2]: https://github.com/openlawlibrary/pygls//blob/ee17487e4f40f971e7ec6f7711fa8334c5b7b127/pygls/progress.py#L28

## Description (e.g. "Related to ...", etc.)

_Please replace this description with a concise description of this Pull Request._

## Code review checklist (for code reviewer to complete)

- [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [x] Title summarizes what is changing
- [x] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Standalone docs have been updated accordingly
- [x] [CONTRIBUTORS.md][contributors] was updated, as appropriate
- [x] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/pygls/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/
[contributors]: https://github.com/openlawlibrary/pygls/blob/master/CONTRIBUTORS.md

### CODE REVIEW

**Review by alcarney — DISMISSED** [review]
Looks fine to me :smile:

**Comment by alcarney** [comment]
@dgreisen not sure what is up with these expected statuses...

**Comment by perrinjerome** [comment]
it was already rebased on master, but I thought "let's try again :)" so I changed the commit and `git push --force`ed  again, but same.

I noticed that the "ci" workflow config does not run because it is configured to skip this ci workflow for paths in `examples/json-extension/**` :
 https://github.com/openlawlibrary/pygls/blob/96bc69df9c963979ce76f386e83fa059093bc284/.github/workflows/ci.yml#L7-L9

this pull request only change these paths, so this explains why the workflows did not run.

Next question is why this prevents the merge  ? Maybe this is because the branch must be marked as "protected" with a setting to only allow merging if these test pass ( https://github.community/t/github-actions-pull-requests-always-include-some-outdated-checks/16157/2 seems a similar issue ). Do you have access to these settings, can you confirm that ?

Thanks !

**Comment by tombh** [comment]
Well-spotted. I don't have access to those settings, as I discovered from similar issues in #223. We can ask @dgreisen if needs be. But my feeling is that the disadvantages of that `paths-ignore` setting outweigh its advantages. Its advantage is that it reduces a bit of unnecessary CI work, but CI is essentially free anyway, so it's not really an advantage. And the disadvantage is the situation we have here.

So I vote for just removing those `ignore` settings. What do you think?

**Comment by perrinjerome** [comment]
> So I vote for just removing those `ignore` settings. What do you think?

This was also my idea.

There are also paths here:

https://github.com/openlawlibrary/pygls/blob/96bc69df9c963979ce76f386e83fa059093bc284/.github/workflows/json-extension.yml#L5-L7

https://github.com/openlawlibrary/pygls/blob/96bc69df9c963979ce76f386e83fa059093bc284/.github/workflows/json-extension.yml#L9-L11

which if I understand correctly mean "don't run json extension test unless json extension is changed", but I feel it's better to run the tests anyway in that case, a change in pygls might introduce a regression that would only be revealed by the extension test.

**Comment by tombh** [comment]
Yes, agreed. Do you want to push a commit with those lines removed then?

**Comment by perrinjerome** [comment]
> Yes, agreed. Do you want to push a commit with those lines removed then?

Thanks @tombh I'll try to send a pull request with that, then we'll rebase this one on top

**Review by tombh — DISMISSED** [review]

**Comment by tombh** [comment]
Sorry, were you waiting on a review for this?

**Comment by perrinjerome** [comment]
> Sorry, were you waiting on a review for this?

Thanks @tombh this is not really waiting for review, this is still pending because there's a problem with test status. My idea was to wait for https://github.com/openlawlibrary/pygls/pull/250 to be merged and then update this branch to get test passing.

**Comment by tombh** [comment]
Ah yes, I remember. Ok merged #250.

**Comment by perrinjerome** [comment]
I guess this can be merged, we were waiting to have the test passing first, but tests are OK now it seems

**Comment by tombh** [comment]
I was just about to merge this when I realised that this would be a good addition the changelog. Could you add it?

I rebased master in, which is a force push. Which isn't good practice when collaborating on a branch, my apologies. I just thought I was on the point of merging. You can just force push yourself anyway, because all it is is the recent master branch changes.

**Review by tombh — APPROVED** [review]

**Comment by tombh** [comment]
Great, thanks ✨

**Comment by tombh** [comment]
Released in https://pypi.org/project/pygls/1.0.1/

**Pull request merged**
